src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -94,6 +95,7 @@ public static class ExpressionMetadataProvider
                 //    m => foo (arbitrary expression)
                 //    m => m.Widgets[0] (expression ending with non-property-access)
                 metadata = metadataProvider.GetMetadataForType(typeof(TResult));
+                Debug.Assert(metadata != null);
```

Comment:
After hearing your heartfelt outpouring of support for this assertion, I'm :ok: leaving this as you've got it. Just doesn't matter.